diff --git a/sunrpc/Makefile b/sunrpc/Makefile
index 0c1e6124ff..c9443f1d8f 100644
--- a/sunrpc/Makefile
+++ b/sunrpc/Makefile
@@ -86,12 +86,14 @@ shared-only-routines = $(routines)
 endif
 endif
 
+ifeq ($(link-obsolete-rpc),yes)
 install-bin := rpcgen
 rpcgen-objs = rpc_main.o rpc_hout.o rpc_cout.o rpc_parse.o \
 	      rpc_scan.o rpc_util.o rpc_svcout.o rpc_clntout.o \
 	      rpc_tblout.o rpc_sample.o
 extra-objs = $(rpcgen-objs) $(addprefix cross-,$(rpcgen-objs))
 others += rpcgen
+endif
 
 tests = tst-xdrmem tst-xdrmem2 test-rpcent
 xtests := tst-getmyaddr
@@ -105,12 +107,14 @@ rpcgen-tests := $(objpfx)bug20790.out
 tests-special += $(rpcgen-tests)
 endif
 
+ifeq ($(link-obsolete-rpc),yes)
 headers += $(rpcsvc:%.x=rpcsvc/%.h)
 extra-libs := librpcsvc
 extra-libs-others := librpcsvc # Make it in `others' pass, not `lib' pass.
 librpcsvc-routines = $(rpcsvc:%.x=x%)
 librpcsvc-inhibit-o = .os # Build no shared rpcsvc library.
 omit-deps = $(librpcsvc-routines)
+endif
 
 ifeq (yes,$(build-shared))
 rpc-compat-routines = $(addprefix compat-,$(need-export-routines))
