diff --git a/misc/sys/cdefs.h b/misc/sys/cdefs.h
index 475cf62961..b29abfeb7b 100644
--- a/misc/sys/cdefs.h
+++ b/misc/sys/cdefs.h
@@ -255,6 +255,13 @@
 # define __nonnull(params)
 #endif
 
+/* GCC 4.3 and above with -std=c99 or -std=gnu99 implements ISO C99
+   inline semantics, unless -fgnu89-inline is used.  */
+#ifdef __GNUC_STDC_INLINE__
+# define __inline __inline __attribute__ ((__gnu_inline__))
+# define __inline__ __inline
+#endif
+
 /* It is possible to compile containing GCC extensions even if GCC is
    run in pedantic mode if the uses are carefully marked using the
    `__extension__' keyword.  But this is not generally available before
