diff --git a/configure b/configure
index aa7869ff17..726efa1798 100755
--- a/configure
+++ b/configure
@@ -7421,7 +7421,7 @@ echo >&5 "libc_undefs='$libc_undefs'"
 # defaults to -fstack-protector but this produces an undefined symbol
 # other than __stack_chk_fail.  However, compilers like that have not
 # been encountered in practice.
-libc_undefs=`echo "$libc_undefs" | egrep '^(foobar|__stack_chk_fail)$'`
+libc_undefs=`echo "$libc_undefs" | sed 's/__stack_chk_fail_local/__stack_chk_fail/' | egrep '^(foobar|__stack_chk_fail)$'`
 case "$libc_undefs" in
 foobar) libc_cv_predef_stack_protector=no ;;
 '__stack_chk_fail
diff --git a/configure.in b/configure.in
index 5028e6411e..a0e4c07079 100644
--- a/configure.in
+++ b/configure.in
@@ -2097,7 +2097,7 @@ echo >&AS_MESSAGE_LOG_FD "libc_undefs='$libc_undefs'"
 # defaults to -fstack-protector but this produces an undefined symbol
 # other than __stack_chk_fail.  However, compilers like that have not
 # been encountered in practice.
-libc_undefs=`echo "$libc_undefs" | egrep '^(foobar|__stack_chk_fail)$'`
+libc_undefs=`echo "$libc_undefs" | sed 's/__stack_chk_fail_local/__stack_chk_fail/' | egrep '^(foobar|__stack_chk_fail)$'`
 case "$libc_undefs" in
 foobar) libc_cv_predef_stack_protector=no ;;
 '__stack_chk_fail
